### PR TITLE
[jormungandr] NewRelic added for stat_manager

### PR DESCRIPTION
**Pull request by azime**
Jira: [NAV-2346](https://navitia.atlassian.net/browse/NAV-2346)

[NAV-2346]: https://navitia.atlassian.net/browse/NAV-2346?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: source/jormungandr/jormungandr/new_relic.py [hunk 1]
```diff
@@ -188,3 +188,39 @@ def record_streetnetwork_call(call_name, connector_name, mode, coverage_name):
...
+            except Exception as e:
+                event_params["status"] = "failed"
+                event_params.update({"reason": str(e)})
+            duration = timeit.default_timer() - start_time
+            event_params.update({"duration": duration})
+
+            # Send the custom event to newrelic !
+            record_custom_event("stat_manager", event_params)
+
+            return result
```
Comment (pbougue) [inline source/jormungandr/jormungandr/new_relic.py]: 2 remarks :
1. Maybe you should also re-raise in the except, not sure.
2. More idiomatic:
```suggestion
                return func(obj, *args, **kwargs)
            except Exception as e:
                event_params["status"] = "failed"
                event_params.update({"reason": str(e)})
            finally:
                duration = timeit.default_timer() - start_time
                event_params.update({"duration": duration})

                # Send the custom event to newrelic !
                record_custom_event("stat_manager", event_params)
```

File: source/jormungandr/jormungandr/stat_manager.py [hunk 2]
```diff
@@ -50,6 +50,10 @@
 f_datetime = "%Y%m%dT%H%M%S"
 
 
+class StatManagerError(RuntimeError):
+    pass
```
Comment (pbougue) [inline source/jormungandr/jormungandr/stat_manager.py]: This should be moved in exceptions.py, no?

File: source/jormungandr/jormungandr/stat_manager.py [hunk 3]
```diff
@@ -556,7 +564,11 @@ def __call__(self, f):
         def wrapper(*args, **kwargs):
             start_time = time.time()
             call_result = f(*args, **kwargs)
-            self.manager.manage_stat(start_time, call_result)
+            try:
+                self.manager.manage_stat(self, start_time, call_result)
+            except (StatManagerError, Exception):
+                # if stat are not working we don't want jormungandr to stop.
+                pass
```
Comment (pbougue) [inline source/jormungandr/jormungandr/stat_manager.py]: StatManagerError is an Exception, so you may just:
```suggestion
            except Exception:
                # if stat are not working we don't want jormungandr to stop.
                pass
```

File: source/jormungandr/jormungandr/stat_manager.py (hunk 2 above)
Comment (azime) [reply source/jormungandr/jormungandr/stat_manager.py]: Done

File: source/jormungandr/jormungandr/stat_manager.py (hunk 3 above)
Comment (azime) [reply source/jormungandr/jormungandr/stat_manager.py]: Done

File: source/jormungandr/jormungandr/new_relic.py (hunk 1 above)
Comment (azime) [reply source/jormungandr/jormungandr/new_relic.py]: Done

**Review by pbougue — APPROVED** [review]

**Pull request merged**